server/src/main/java/org/apache/iotdb/db/tools/virtualSG/DeviceMappingViewer.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.tools.virtualSG;
+
+import java.util.Set;
+import org.apache.iotdb.db.conf.IoTDBDescriptor;
+import org.apache.iotdb.db.engine.storagegroup.virtualSg.HashVirtualPartitioner;
+import org.apache.iotdb.db.exception.metadata.MetadataException;
+import org.apache.iotdb.db.metadata.MManager;
+import org.apache.iotdb.db.metadata.PartialPath;
+
+/**
+ * for DBA to view the mapping from device to virtual storage group ID
+ * usage: run this class with arguments [system_schema_dir], if args are not given, use default in config
+ */
+public class DeviceMappingViewer {
+
+  public static void main(String[] args) throws MetadataException {
+    // has schema log dir
+    if(args.length == 1){
+      IoTDBDescriptor.getInstance().getConfig().setSchemaDir(args[0]);
+    }
+
+    HashVirtualPartitioner partitioner = HashVirtualPartitioner.getInstance();
+    IoTDBDescriptor.getInstance().getConfig().setEnableMTreeSnapshot(false);
+    MManager mManager = MManager.getInstance();
+    mManager.init();
```

Comment:
`MManager.init()` initializes writers, so it may throw an IOException if it is used against a working IoTDB instance because the files (mlog and tlog) have already been opened by that IoTDB.
Maybe a read-only mode of `MManager` is necessary.

Reply:
Mac and linux can let 2 writers to write one file. Let me test it on windows~

Reply:
I have test it on windows and it's OK~